lib/src/iterable_extensions.dart

```diff
@@ -589,6 +589,19 @@ extension IterableExtension<T> on Iterable<T> {
     }
     return true;
   }
+
+  /// The elements with [seperator] between each.
+  Iterable<T> seperatedBy(T seperator) sync* {
+    if (isEmpty) return;
+
+    final iterator = this.iterator;
+    iterator.moveNext();
```

Comment:
Should check whether this returns false, and return if it does, instead of having a separate `isEmpty` at the top. (In general, it's best to not nterate the same iterable more than once).